test-fixture/src/sim/taildrop.rs

```diff
@@ -7,20 +7,107 @@
 #![expect(clippy::unwrap_used, reason = "This is test code.")]
 
 use std::{
-    cmp::max,
+    cmp::{max, min},
     collections::VecDeque,
-    fmt::{self, Debug},
+    fmt::{self, Debug, Display},
     time::{Duration, Instant},
 };
 
-use neqo_common::{qinfo, qtrace, Datagram};
+use neqo_common::{qinfo, qtrace, Datagram, Dscp, Ecn, Tos};
 use neqo_transport::Output;
 
 use super::Node;
+use crate::sim::Rng;
 
 /// One second in nanoseconds.
 const ONE_SECOND_NS: u128 = 1_000_000_000;
 
+#[derive(Clone)]
+struct Stats {
+    /// The number of packets received.
+    received: usize,
+    /// The number of packets marked.
+    marked: usize,
+    /// The number of packets dropped.
+    dropped: usize,
+    /// The number of packets delivered.
+    delivered: usize,
+    /// The maximum amount of queue capacity ever used.
+    /// As packets leave the queue as soon as they start being used, this doesn't
+    /// count them.
+    maxq: usize,
+    /// The buffer capacity.
+    capacity: usize,
+
+    /// The start time.
+    start_time: Option<Instant>,
+    /// The start time.
+    last_update: Option<Instant>,
+    /// Accumulated queue usage multiplied by time.
+    /// This is in units of `bytes * nanoseconds`, which will get big.
+    /// For a test that transfers 1 gigabyte and takes a month,
+    /// that's still well within 64 bits.  Use `u128` to be safe.
+    cumulative_usage: u128,
+}
+
+impl Stats {
+    const fn new() -> Self {
+        Self {
+            received: 0,
+            marked: 0,
+            dropped: 0,
+            delivered: 0,
+            maxq: 0,
+            capacity: 0,
+            start_time: None,
+            last_update: None,
+            cumulative_usage: 0,
+        }
+    }
```

Comment:
Derive `default` and use it instead?

Reply:
`Default` isn't `const`, which turned out to be a necessity for some reason.  (I think that it's because of how we set up the simulator tests with macros.)